src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentImportProjectFeature.cs

```diff
@@ -99,6 +97,8 @@ public VirtualProjectItem(string content)
 
             public override bool Exists => true;
 
+            public override string FileKind => FileKinds.ComponentImport;
```

Comment:
This is not being used yet. I plan to use this to add errors for adding code, markup etc in an import document,

Reply:
It might make more sense to use a document classifier for this. FileKind is something that has to be represented in MSBuild.

Reply:
I see what you mean. But document classification happens after parsing and IR lowering. I was hoping to add those errors at one of those layers. This means I'll have to do this in a separate IR pass.

Reply:
OK thats fine - add this logic to the SDK as well (with tests). Ultimately we want the project system (and msbuild) to tell us how to interpret the files.

Reply:
After thinking about this more, I think I agree with you that FileKind is the right choice.